reana_client/cli/files.py

```diff
@@ -247,16 +250,27 @@ def download_files(
                     )
                 )
 
-                outputs_file_path = os.path.join(output_directory, file_name)
-                if not os.path.exists(os.path.dirname(outputs_file_path)):
-                    os.makedirs(os.path.dirname(outputs_file_path))
+                if output_directory == "-":
+                    if multiple_files_zipped:
```

Comment:
If it's a zip file, ie. multiple files to display, it would be nice to display each filename before its contents. Right now everything is mixed up so it's not easy to know which content corresponds to which file.

For example, executing this command in the roofit example returns the following:
```console
$ reana-client download -w roofit 'code/' -o -
#ifndef __CINT__                                                      
#include "RooGlobalFunc.h"                                            
#endif                                                                                                                                      
#include "RooRealVar.h"                                               
#include "RooDataSet.h"                                               
#include "RooGaussian.h"                                              
#include "RooChebychev.h"                                             
#include "RooAddPdf.h"                                                
#include "RooExtendPdf.h"                                             
#include "TCanvas.h"   
....
#ifndef __CINT__                                                      
#include "RooGlobalFunc.h"                                            
#endif                                                                
#include "RooRealVar.h"                                               
#include "RooDataSet.h"                                               
#include "RooGaussian.h"                                              
#include "RooChebychev.h"                                             
#include "RooAddPdf.h"                                                
#include "RooExtendPdf.h"                                             
#include "TCanvas.h"           
...
```

which corresponds to the 2 files that are in the `code` folder. However I don't know what's the correspondence of each content.

```console
$ reana-client ls -w roofit
NAME                SIZE     LAST-MODIFIED      
code/fitdata.C      1648     2022-02-17T10:43:09
code/gendata.C      1937     2022-02-17T10:43:09
results/data.root   154458   2022-02-17T10:43:23
results/plot.png    15450    2022-02-17T10:43:29
```

Maybe we can add some headers to differentiate.

```
==> File: code/fitdata.C
```

Not sure if this is the intention of the task though, or if we just want the raw content. But as a user is hard to understand the content of multiple files at the same time. cc/ @tiborsimko

Reply:
I agree that showing an header for each file would make the output more clear. However, printing only the raw content of the files makes it possible to pipe the output of `reana-client` to another command line utility, for example:
```bash
reana-client download -o - results.json | jq ...
```
This is probably only interesting when dealing with a single file, so perhaps we could only show the headers when multiple files are being displayed?

Reply:
> This is probably only interesting when dealing with a single file, so perhaps we could only show the headers when multiple files are being displayed?

Definitely, that was my point :) hence my "If it's a zip file,...". I should have explained myself better.

Reply:
The goal of the issue was to emulate Unix `cat`, see the original issue description. I would go for simply concatenating the output files without any in-between headers. In this way, we can say that outputting to stdout is doing exactly what Unix `cat` is doing, and users will understand the behaviour by that analogy. That should do for MVP.